encrypt-config/ConfigFile.h

```diff
@@ -0,0 +1,85 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include "utils/EncryptionUtils.h"
+#include "utils/OptionalUtils.h"
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace encrypt_config {
+
+class ConfigLine {
+public:
+  ConfigLine(std::string line);
+  ConfigLine(const std::string& key, const std::string& value);
+
+  void updateValue(const std::string& value);
+
+  std::string line_;
+  std::string key_;
+  std::string value_;
```

Comment:
These should be private because there is an invariant to be enforced. https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rc-struct

Alternatively, we should not keep the original line, only the key and value. In this case, the class essentially becomes `std::pair<std::string, std::string>`.

```suggestion
  std::string key() const { return key_; }
  std::string value() const { return value_; }
 private:
  std::string line_;
  std::string key_;
  std::string value_;
```

Reply:
We need the original line, because I do not want to reformat the line unless we modify it.  For example, if there are no properties to be encrypted, `encrypt-config` should be a no-op.

Reply:
Could you add a comment about this design decision? It will help future readers.